src/ThermalCameraSensor.cc

```diff
@@ -422,9 +462,29 @@ bool ThermalCameraSensor::Update(
   frame->add_value(this->Name());
 
   std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
-  this->dataPtr->thermalMsg.set_data(this->dataPtr->thermalBuffer,
-      rendering::PixelUtil::MemorySize(rendering::PF_L16,
-      width, height));
+
+  // \todo(anyone) once ign-rendering supports an image event with unsigned char
+  // data type, we can remove this check that copies uint16_t data to char array
+  if (this->dataPtr->thermalCamera->ImageFormat() == rendering::PF_L8)
+  {
+    unsigned int len = width * height;
+    if (!this->dataPtr->thermalBuffer8Bit)
+      this->dataPtr->thermalBuffer8Bit = new unsigned char[len];
+    for (unsigned int i = 0; i < len; ++i)
+    {
+      this->dataPtr->thermalBuffer8Bit[i] =
+          static_cast<unsigned int>(this->dataPtr->thermalBuffer[i]);
```

Comment:
I'm not sure that I understand this completely. It looks like we are assigning an `unsigned int` to an `unsigned char`. Could there by chance be any data that would be truncated here since `unsigned int` has more bytes than `unsigned char`, or is the data in the `unsigned int` already in the least-most/right-most byte before performing this assignment?

Reply:
211d16f65902b913b6a226d5bade1792f7078fce, now using `uint8_t`